R/decks.R

```diff
@@ -0,0 +1,482 @@
+setMethod("decks", "CrunchDataset", function (x) {
+    DeckCatalog(crGET(shojiURL(x, "catalogs", "decks")))
+})
+
+
+# Deck Catalog ------------------------------------------------------------
+
+#' Create an empty Crunch Deck
+#'
+#' @param dataset A Crunch Dataset
+#' @param name The name of the Deck
+#' @param ... Further attributes of the deck such as the description, see API
+#'   docs for options.
+#' @return The `CrunchDeck` that was created.
+#' @export
+newDeck <- function(dataset, name, ...) {
+    stopifnot(is.dataset(dataset))
+    payload <- wrapEntity(name = name, ...)
+    url <- shojiURL(dataset, "catalogs", "decks")
+    url <- crPOST(url, body = toJSON(payload))
+    return(CrunchDeck(crGET(url)))
+}
+
+setMethod("initialize", "DeckCatalog", init.sortCatalog)
+
+setMethod("[[", "DeckCatalog",  function (x, i, ...) {
+    getEntity(x, i, CrunchDeck, ...)
+})
+
+setMethod("[[", c("DeckCatalog", "character", "ANY"),  function (x, i, ...) {
+    if (length(i) > 1) {
+        halt("You can only select one deck at a time")
+    }
+    matches <- names(x) %in% i
+    if (all(!matches)) {
+        halt(dQuote(i), " is not present in deck catalog")
+    }
+    index <- which(matches)
+    if (sum(matches) > 1) {
+        warning(dQuote(i), " does not uniquely identify elements. Returning the first match")
+        index <- index[1]
+    }
+    getEntity(x, index, CrunchDeck, ...)
+})
+
+setMethod("show", "DeckCatalog", function(object){
```

Comment:
The `as.data.frame` method works on deck catalogs, I think the reason I defined a different show method was just to not print the `creation_time` and `owner_id` fields which take up a lot of screen real estate. But maybe it's better to just print everything instead of doing that. What do you think?

Reply:
I don't have a strong opinion either way. For some other options/patterns, see https://github.com/Crunch-io/rcrunch/blob/master/R/show.R (and search for "Catalog").